src/malli/core.cljc

```diff
@@ -1307,7 +1307,7 @@
         ^{:type ::schema}
         (reify
           AST
-          (-to-ast [_ _] {:type :enum, :values children})
+          (-to-ast [_ _] (-ast {:type :enum :values children} properties options))
```

Comment:
I wonder if you need to call `(ast children` the same way that [`-map-schema`](https://github.com/rralston-sfin/malli/blob/9dfc0a97d1bfac9636636c3a4950d9a6a12bb609/src/malli/core.cljc#L1089) does?

Reply:
I don't think so. The children on a map can be nested schemas so you need to traverse the tree. The `-enum-schema` validator is just `contains?` so I don't think the children can be schemas.